actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -92,10 +93,27 @@ def redirect_back(fallback_location:, allow_other_host: true, **args)
     #
     # All other options that can be passed to #redirect_to are accepted as
     # options and the behavior is identical.
-    def redirect_back_or_to(fallback_location, allow_other_host: true, **args)
-      referer = request.headers["Referer"]
-      redirect_to_referer = referer && (allow_other_host || _url_host_allowed?(referer))
-      redirect_to redirect_to_referer ? referer : fallback_location, **args
+    def redirect_back_or_to(fallback_location, allow_other_host: _allow_other_host, **options)
+      referer_location = request.referer
+      redirect_to_referer = referer_location && (allow_other_host || _url_host_allowed?(referer_location))
+      location = redirect_to_referer ? referer_location : fallback_location
+
+      allow_other_host = true if _allow_other_host && !allow_other_host # if the fallback is an open redirect
```

Comment:
I think I'm missing the point of having a private `_redirect_to`. Right now you could call public `redirect_to` and it would make no difference. I tried for awhile to extract URL computation but I couldn't find a good way of doing it that didn't require multiple computations / perf regressions. 🙄 

I imagine if I get it right I will not need this line.